compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/optimizer/BIRBasicBlockOptimizer.java

```diff
@@ -73,6 +69,7 @@ public void visit(BIRNode.BIRTypeDefinition birTypeDefinition) {
 
     @Override
     public void visit(BIRNode.BIRFunction birFunction) {
+        BIRGenUtils.rearrangeBasicBlocks(birFunction);
```

Comment:
Why are we calling `rearrangeBasicBlocks()` twice here?

Reply:
At first we need to rearrange to check whether there are any loops in basicblocks to make sure not to remove them during optimiser.

Second we need to rearrange to properly show them in bir dump.